pkg/initializer_v2/test/unit/dataset/test_dataset.py

```diff
@@ -0,0 +1,86 @@
+import runpy
```

Comment:
Why do we want to use runpy to execute the tests ?

Reply:
I use `runpy` to run the [dataset](https://github.com/kubeflow/training-operator/blob/master/pkg/initializer_v2/model/__main__.py#L14), and [model](https://github.com/kubeflow/training-operator/blob/master/pkg/initializer_v2/dataset/__main__.py) modules.

Reply:
Do you really need to execute `__main__.py` as part of your unit tests ?
E.g. the entire logic can be tested in the `dataset/huggingface_test.py`  and `model/huggingface_test.py` 

I guess, you can verify that `__main__.py` executes correctly as part of your E2E tests.
WDYT @seanlaii ?

Reply:
Yes, I can verify the `__main__.py` in the E2E tests.
Or perhaps we can wrap the logic in the script to a function, e.g., `main()`, so we can avoid using `runpy` to run the script, and just validate the `main()` function in `huggingface_test.py`.
The reason I try to execute the script is mainly to validate the overall flow and some exceptions.

Reply:
I see, that makes sense. Maybe we should wrap our logic under `main()` func in the `__main__.py` file given that usually `runpy` is used for integration tests, not for unit testing.
So we can have this. `__main__.py`:
```python
def main():
  # logic here.

if __name__ == "__main__":
  main()
```

`main_test.py`: Contains unit tests for the `main()` function.

WDYT @seanlaii ?

Reply:
Yes, it sounds good to me.